Make data sources use `AppLayoutSimpleSaveCancelTitle` + full-screen

spolu opened the pull request:
Fixes #1752 

Basically more sparkle-ization

spolu commented:
It's a bit heavy but there's not much in there. Mostly UI stuff.

r? the readOnly / admin / user stuff are not changed
Maybe test it to convice yourself :grimacing:

PopDaph commented:
I have a JS error when going on one of my static datasources page, not all the time tho. Do you reproduce? 
<img width="897" alt="Capture d’écran 2023-09-27 à 17 03 37" src="https://github.com/dust-tt/dust/assets/3803406/fe7baa00-9eab-48f2-b67a-c50325152ac2">

PopDaph commented on front/pages/w/[wId]/builder/data-sources/new.tsx:
```diff
@@ -59,8 +60,9 @@ export default function DataSourceNew({
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
-  const [disabled, setDisabled] = useState(true);
-  const [creating, setCreating] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+  const [edited, setEdited] = useState(false);
+  const [valid, setValid] = useState(true);
```
Nit consistency: adding `is` prefix for all or for none? (I vote for having the prefix so we instantly know it's a boolean)

spolu replied:
Done :+1:

PopDaph commented on front/pages/w/[wId]/builder/data-sources/new.tsx:
```diff
@@ -78,35 +83,51 @@ export default function DataSourceNew({
-      return false;
+      valid = false;
     } else {
+      edited = true;
```
NIT: initiate valid to false, and set it here to true instead of in all if??

spolu replied:
That does not scale to having other field being possibly invalid?

PopDaph commented on front/pages/w/[wId]/builder/data-sources/new.tsx:
```diff
@@ -70,6 +72,9 @@ export default function DataSourceNew({
   const [dataSourceDescription, setDataSourceDescription] = useState("");
 
   const formValidation = useCallback(() => {
+    let edited = false;
```
NIT: shouldn't `edited` be set to true as long as there was a change on any field, so directly on the onChange methods? I feel it's slightly weird to keep a edited = false if datasource name exists already or starts with manage-.

spolu replied:
Well you want edited to be false if the value matches the original value, so it's kind of nicer to have that logic centralized?

PopDaph reviewed (approved):
Just a few NITs, feel free to discard them. 
(Just maybe check if you also have the JS error on the static datasource page?)

spolu commented:
@PopDaph fixed the error that predated that PR (nice catch)
Also addressed some of your comments in the last commit :+1: 
WIll merge as is unless I hear otherwise :pray:
